python/pyspark/java_gateway.py

```diff
@@ -33,14 +33,18 @@
 from pyspark.serializers import read_int
 
 
-def launch_gateway(conf=None):
+def launch_gateway(conf=None, popen_kwargs=None):
```

Comment:
I'd make this _popen_kwargs to indicate it's usage is possibly not super supported.

Reply:
Would a comment in the docstring to that effect be better? I haven't seen `_var_name` used in Python projects to indicate a developer feature. (But of course, maybe I've just not seen it yet!)